### ISSUE TITLE
Row style bottom margin doesn't apply to widgets when there are more than one widget in a column

### ISSUE THREAD

**Issue by AlexGStapleton** [siteorigin/siteorigin-panels#250]
As the title suggests if you set a bottom margin of say, 0 and there are multiple widgets in a column the bottom margin will only apply to the last widget. All other widgets will have the global bottom margin default rather than the desired bottom margin of 0. The no bottom margin setting (in the layout settings group) has no effect either.

Here my [Example layout](https://u.teknik.io/dEOjA.json) which is used in all of the below screenshots.

Screenshot of bottom margin being applied to widget in a multi-widget column.
![](https://i.imgur.com/zME0YuR.png)

No bottom margin being applied to the last widget in that column:
![](https://i.imgur.com/TqZFG6M.png)

No margin being applied in the column to the left when no other widgets are present:
![](https://i.imgur.com/ErodBdx.png)

Adjusted global bottom margin (from 35 to 10) setting and change was reflected:
![](https://i.imgur.com/aBwL5e2.png)

**Comment by AlexGStapleton** [comment]
As this has been merged in PB 3.0 I'm going to close this for the time being.

**Comment by AlexGStapleton** [comment]
Re-opened as PB 3.0 was changed to 2.5 and this wasn't included.

**Comment by braamgenis** [comment]
It sounds like things are working as they should. _Row_ styles should only affect _rows_. It seems we don't currently have a way to set margins on individual cells/widgets, but we should add it if that's a commonly requested feature.

**Comment by AlexGStapleton** [comment]
I agree, but this issue is more about how things currently work (everything based on row bottom margin) rather than how they should work (being the separated dedicated row and widget bottom margins).

Widget bottom margins are very commonly requested. Cells, not so much (I'm struggling to think of a request).
